### PR TITLE
Add `list project environment variable` subcommand (one of commands to manage projects)

### PR DESCRIPTION
**Pull request by threepipes**
# Checklist

=========

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked for similar issues and haven't found anything relevant.
  - https://github.com/CircleCI-Public/circleci-cli/issues/652 is the issue that this PR is targeting
- [x] This is not a security issue (which should be reported here: https://circleci.com/security/)
- [x] I have read [Contribution Guidelines](https://github.com/CircleCI-Public/circleci-cli/blob/master/CONTRIBUTING.md).

## Changes

=======

- Add a project command
  - Add a subcommand to list project environment variables
    - `circleci project secret list <vcs> <org> <project>`

## Rationale

=========

This PR partially solves https://github.com/CircleCI-Public/circleci-cli/issues/652.
Currently, CircleCI CLI does not have the ability to manage Projects.
This change enables one of these abilities and makes it easier to write the rest subcommands.

## Considerations

==============

I will add other subcommands to manage project environment variables like `create` and `delete`.

## **Here are some helpful tips you can follow when submitting a pull request:**

1. Fork [the repository](https://github.com/CircleCI-Public/circleci-cli) and create your branch from `master`.
2. Run `make build` in the repository root.
3. If you've fixed a bug or added code that should be tested, add tests!
4. Ensure the test suite passes (`make test`).
5. The `--debug` flag is often helpful for debugging HTTP client requests and responses.
6. Format your code with [gofmt](https://golang.org/cmd/gofmt/).
7. Make sure your code lints (`make lint`). Note: This requires Docker to run inside a local job.

### CODE REVIEW

File: api/rest/client.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func (c *Client) NewRequest(method string, u *url.URL, payload interface{}) (req
 	}
 
 	req.Header.Set("Circle-Token", c.circleToken)
-	req.Header.Set("Accept-Type", "application/json")
+	req.Header.Set("Accept", "application/json")
```
Comment (threepipes) [inline api/rest/client.go:61]: I wrote a description of this change in the `Consideration` section of the PR description.

**Comment by corinnesollows** [comment]
Hello @threepipes This PR is large, could it possibly be broken down into smaller PRs. ie: Accept-Type in one PR and project command in a separate one? Thank you

**Comment by threepipes** [comment]
The header changes are extracted: https://github.com/CircleCI-Public/circleci-cli/pull/784

**Comment by threepipes** [comment]
@corinnesollows 
Thank you for your approval of https://github.com/CircleCI-Public/circleci-cli/pull/784.
This PR no longer contains the `Accept-Type` header changes.
If there are any more requirements, feel free to let me know.

**Comment by flanger001** [comment]
It would be extremely tight if this could get some attention @CircleCI-Public/Extensibility @corinnesollows

**Review by JulesFaucherre — APPROVED** [review]

**Pull request merged**